webviz_ert/controllers/plot_view_controller.py

```diff
@@ -11,79 +11,77 @@
+    elif data_type == DataType.RESPONSE:
+        webviz_ert.controllers.multi_response_controller(parent, app)
+    else:
+        raise Exception(f"Unexpected data type `{data_type.value}`")
 
     @app.callback(
-        Output(parent.uuid("plot-selection-store"), "data"),
+        Output(parent.uuid(f"plot-selection-store-{data_type.value}"), "data"),
         [
-            Input(parent.uuid("parameter-selection-store-param"), "modified_timestamp"),
-            Input(parent.uuid("parameter-selection-store-resp"), "modified_timestamp"),
+            Input(
+                parent.uuid(f"parameter-selection-store-{data_type.value}"),
+                "modified_timestamp",
+            ),
         ],
         [
-            State(parent.uuid("parameter-selection-store-param"), "data"),
-            State(parent.uuid("parameter-selection-store-resp"), "data"),
-            State(parent.uuid("plot-selection-store"), "data"),
+            State(parent.uuid(f"parameter-selection-store-{data_type.value}"), "data"),
+            State(parent.uuid(f"plot-selection-store-{data_type.value}"), "data"),
         ],
     )
     def update_plot_selection(
         _: Any,
-        __: Any,
-        parameters: Optional[List[str]],
-        responses: Optional[List[str]],
-        current_selection: Optional[List[Mapping[str, str]]],
-    ) -> List[Mapping[str, str]]:
-        parameters = [] if not parameters else parameters
-        responses = [] if not responses else responses
-        current_selection = [] if not current_selection else current_selection
-        for plot in current_selection.copy():
-            if plot["name"] not in parameters and plot["name"] not in responses:
-                current_selection.remove(plot)
-        for param in parameters:
-            new_plot = {"name": param, "type": "parameter"}
-            if new_plot not in current_selection:
-                current_selection.append(new_plot)
-        for response in responses:
-            new_plot = {"name": response, "type": "response"}
-            if new_plot not in current_selection:
-                current_selection.append(new_plot)
-
-        return current_selection
+        selection: Optional[List[str]],
+        current_plots: Optional[List[str]],
+    ) -> List[str]:
+        selection = [] if not selection else selection
+        current_plots = [] if not current_plots else current_plots
+        for plot in current_plots.copy():
+            if plot not in selection:
+                current_plots.remove(plot)
+        for selected in selection:
+            if selected not in current_plots:
+                current_plots.append(selected)
+
+        return current_plots
 
     @app.callback(
-        Output(parent.uuid("plotting-content"), "children"),
-        Input(parent.uuid("plot-selection-store"), "data"),
-        State(parent.uuid("plotting-content"), "children"),
+        Output(parent.uuid(f"plotting-content-{data_type.value}"), "children"),
```

Comment:
I don't think you need {data_type.value} I think {data_type} should be enough here.

Reply:
ya awesome indeed! i must have misunderstood the docs... thanks!
fixed.